crates/ruff/src/rules/pylint/rules/invalid_string_characters.rs

```diff
@@ -0,0 +1,205 @@
+use rustpython_parser::ast::Location;
+
+use ruff_macros::{derive_message_formats, violation};
+
+use ruff_diagnostics::AlwaysAutofixableViolation;
+use ruff_diagnostics::Fix;
+use ruff_diagnostics::{Diagnostic, DiagnosticKind};
+use ruff_python_ast::source_code::Locator;
+use ruff_python_ast::types::Range;
+
+/// ## What it does
+/// Checks that strings don't have the control character BS
+///
+/// ## Why is this bad?
+/// Control characters can display differently in different text editors and terminals. By using the \b sequence the string's
+/// value will be the same but it will be visible in all text editors.
+///
+/// ## Example
+/// ```python
+/// x = ''
+/// ```
+///
+/// Use instead:
+/// ```python
+/// x = '\b'
+/// ```
+#[violation]
+pub struct InvalidCharacterBackspace;
+
+impl AlwaysAutofixableViolation for InvalidCharacterBackspace {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Invalid unescaped character backspace, use \"\\b\" instead")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Replace with escape sequence".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks that strings don't have the raw control character SUB
+///
+/// ## Why is this bad?
+/// Control characters can display differently in different text editors and terminals. By using the \x1B sequence the string's
+/// value will be the same but it will be visible in all text editors.
+///
+/// ## Example
+/// ```python
+/// x = ''
+/// ```
+///
+/// Use instead:
+/// ```python
+/// x = '\x1A'
+/// ```
+
+#[violation]
+pub struct InvalidCharacterSub;
+
+impl AlwaysAutofixableViolation for InvalidCharacterSub {
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Invalid unescaped character SUB, use \"\\x1A\" instead")
+    }
+
+    fn autofix_title(&self) -> String {
+        "Replace with escape sequence".to_string()
+    }
+}
+
+/// ## What it does
+/// Checks that strings don't have the raw control character ESC
+///
+/// ## Why is this bad?
+/// Control characters can display differently in different text editors and terminals. By using the \x1B sequence the string's
+/// value will be the same but it will be visible in all text editors.
+///
+/// ## Example
+/// ```python
+/// x = ''
+/// ```
+///
+/// Use instead:
+/// ```python
+/// x = '\x1B'
+/// ```
+
```

Comment:
We prefer not to have an empty line between the doc comment and a type's attributes.

```suggestion
```